accepted/00000-remote-command-auditing.md

```diff
@@ -0,0 +1,116 @@
+- A `Remote Command` tab is present in the details section of a traditional or minion system (or a `System Set Manager` group)
+- The API exposes the `system.scheduleScriptRun` to issue a command
+- (Only for minions) a `Salt > Remote Commands` page to launch a command on minions by leveraging [Salt targeting](https://docs.saltstack.com/en/latest/topics/targeting/index.html) to identify the target(s)
+
+There are the following rules already in place:
+
+- Every user that with the proper role can access the tab and run a remote command on the selected system or a group of systems
+- For every user registered to SUSE Manager, independent of the role, the menu item `Salt > Remote Commands` is accessible. Only the users that have the proper role can issue any command targeting the systems he/she is entitled to access with his/her role.
+- The API is always exposed but checks for the role of the user issuing the command after the API is called
+
+How roles profile the systems that a user can reach?
+
+A user associated with the following roles can issue any command in the associated systems:
+
+- Users associated with any of the `Administrative Roles` (`SUSE Manager Administrator` or `Organization Administrator`) can reach any system within the same organization
+- Every other user (from now on: one does not have any of the above roles) can reach all the systems that he is been assigned to in his/her `System Groups` (this is regulated in the `rhnuserserverperms` table)
+
+NOTE: starting from SUSE Manager 4.0, every command issued in the `Salt > Remote Commands` page is logged.
+
+## Proposal
+
+The motivations behind the changes introduced in this RFC are _security_ and _auditing_. Any command issues via the `Remote Command` will be run with `root` privileges on the target system(s).
+
+To overcome the above limitations, in this RFC we are going to introduce a new role for enabling/disabling the `Remote Command(s)` feature for users that do not have `Administrative Roles`.
+
+Users associated with `Administrative Roles` will not be impacted: they are assimilated as being `root` and they are allowed to do anything.
+
+## Caveats
+
+From a security point of view, if a user does not have any role associated to him/her the surface of the attack is still not negligible: the user can install and remove packages using SUSE Manager on the systems he/she is assigned to. Those packages can be ad-hoc crafted to contain {pre, post}-install script to execute what an attacker wants.
+
+### Comparison with `Configuration Administrator`
+
+A user having the `Configuration Administrator` role can create and apply states to the systems he/she is entitled to. This offers the same surface of attack, as an attacker could simply drop a `cmd.run` into the desired state.
+
+Should `Remote Command Administrator` just another addition to the access control list of the `Configuration Administrator`?
```

Comment:
I am not sure I get the meaning of this question. Are you suggesting that Configuration Administrator implies Remote Command Administrator?

Reply:
I think these should be 2 different roles. A "configuration administrator" should only be able to create of change configuration channels. It could mean that he has no access to the system and should not be able to run any command.

Reply:
> I am not sure I get the meaning of this question. Are you suggesting that Configuration Administrator implies Remote Command Administrator?

Yes, given that a `Configuration Administrator` can apply states and execute them, why not assimilate `Remote Command Administrator` into the `Configuration Administrator`?

Reply:
> I think these should be 2 different roles. A "configuration administrator" should only be able to create of change configuration channels. It could mean that he has no access to the system and should not be able to run any command.

This could be another possibility. In other words: a `Configuration Administrator` only writes states, but cannot apply them. A `Remote Command Administrator` can apply states and issue commands. Did I get it right?

Reply:
I fear we are currently mixing here the current situation with future expectations.

I think we understood the current permission system totally wrong. All currently existing administrator roles are **not** for Systems. They are only for management operation inside of SUSE Manager.

This means the **Configuration Administrator** can only create and modify config channels and states, but he cannot call state.apply. We should prove this. We all have not enough played around with the permission system to know this for sure.

Another "guess" we have is: Any normal user can become administrator of a system and with this permissions he is allowed to do everything with that system. Such a user is "root".
There is no special "Role" for this.

It seems that people want to change this. But this become much harder then expected.
This means we need to split the functionallity of "root" into pieces which makes sense.

In such a case we need to create table of new roles with functionalities in the systems page.

Reply:
I agree with MC.

To your comment regarding remote command administrator. 
A system administrator can execute states or perform a highstate. Only when he also has the role remote command administrator he can execute remote commands for the systems he is administrator.